### PR TITLE
feature: Store the username in eventlog for any entries created through the Webui

### PR DESCRIPTION
**Pull request by towster**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

**Comment by laf** [comment]
Great idea - I'd personally say we should add another column and record this properly. @librenms/reviewers thoughts?

**Comment by geordish** [comment]
Great idea, but I agree with @laf. A column on the event to relate it back to a person would be much better.

**Comment by laf** [comment]
I've updated this now to store username. Chose not to do user_id due to varying auth methods and how inconsistence this could be. @librenms/reviewers can someone review pls.

**Comment by towster** [comment]
Thanks for doing this more properly laf

**Comment by laf** [comment]
No worries, thanks for prompting us into it :)

**Comment by laf** [comment]
Updated schema file.

File: html/includes/table/eventlog.inc.php [hunk 1]
```diff
@@ -60,12 +60,16 @@
         $type = $eventlog['type'];
     }
     $severity_colour = $eventlog['severity'];
+    if ($eventlog['username'] === 0) {
```
Comment (murrant) [inline html/includes/table/eventlog.inc.php]: Are you sure the identical comparison (===) is the correct one here?
Aren't all db results strings?

File: includes/functions.php [hunk 2]
```diff
@@ -830,13 +830,19 @@ function log_event($text, $device = null, $type = null, $severity = 2, $referenc
         $device = device_by_id_cache($device);
     }
 
+    $username = '';
+    if (function_exists(get_userid) && !empty($_SESSION['username'])) {
```
Comment (murrant) [inline includes/functions.php]: function_exists() a left over?

Comment (laf) [inline includes/functions.php]: Yeah I had userid as the one being stored originally.

Updated

**Pull request merged**